src/signed_data.rs

```diff
@@ -145,6 +145,12 @@ pub fn verify_signature(signature_alg: &SignatureAlgorithm,
         .map_err(|_| Error::InvalidSignatureForPublicKey)
 }
 
+pub fn public_key_bytes(spki_value: untrusted::Input) ->
```

Comment:
I think it would make more sense to have a function that returns a `SubjectPublicKeyInfo(&'a [u8])` struct that references the entire SPKI, and which implements `AsRef<[u8]>`, and then have that SPKI structure have a `subjectPublicKey()` function that returns a `SubjectPublicKey(&'a [u8])` that references the value of the key and which also implements `AsRef<[u8]>`).

Reply:
One thing I wonder is, does it not make sense to use `SubjectPublicKeyInfo` (the type already existing in `signed_data.rs`) directly inside `Cert`?

Reply:
> One thing I wonder is, does it not make sense to use `SubjectPublicKeyInfo` (the type already existing in `signed_data.rs`) directly inside `Cert`?
> 
> [Edit] Ah no, I guess that makes for example trust_anchor_from_cert harder.

Yes, I think that is a good idea. See #92, which stores the entire `SubjectPublicKeyInfo` structure (including tag and length) for future use. It does it using the type `der::Value`. It would be better to change this PR to change the type of that field from `der::Value` to `SubjectPublicKeyInfoRef`.

Reply:
So should I wait for #92 to be merged and then rebase this on top of it?

Reply:
> So should I wait for #92 to be merged and then rebase this on top of it?

Do you think #92 would be helpful for this? I did write it with this in mind.

Do you think, instead of `SubjectPublicKeyInfoRef`, we could have `SubjectPublicKeyInfo<A: AsRef<[u8]>`, where `SubjectPublicKeyInfo<&[u8]>` would work like `SubjectPublicKeyInfoRef`, and also `SubjectPublicKeyInfo<Vec<u8>>` could be allowed? (And similarly for `SubjectPublicKey`.)

I don't think it's great to `impl From<X> for untrusted::Input` for these types because it is ambiguous what bytes would be in the result. For example, would they include the tag and length part or not? I would rather have explicitly-named methods that indicate what is returned. WDYT?

Reply:
Hi, sorry for taking so long to reply. I have thought if over a lot but I am not 100% certain I understand what you're getting at, so I'll try to reply to the individual points.

First, getting rid of untrusted for `SubjectPublicKeyInfo` seems good and in line with other API changes that have been done, so that I definitely agree with.

Second, I think allowing owned values does make total sense. Do you envision an API where we would also export a `SubjectPublicKeyInfo<Vec<u8>>`, `to_owned()` or so?

Third, not implementing `From` is fine for me, it just seemed like an obvious conversion to me - but that was probably driven only by my use case, not a more broad evaluation of what this could be used for.

Finally, having "pre-parsed" values inside `Cert` seems quite sensible, so I agree to an approach as in #92 or maybe even more aggressive for other fields as well.